pkg/services/sqlstore/user_auth.go

```
@@ -0,0 +1,133 @@
+package sqlstore
+
+import (
+	"time"
+
+	"github.com/grafana/grafana/pkg/bus"
+	m "github.com/grafana/grafana/pkg/models"
+)
+
+func init() {
+	bus.AddHandler("sql", GetUserByAuthInfo)
+	bus.AddHandler("sql", GetAuthInfo)
+	bus.AddHandler("sql", SetAuthInfo)
+	bus.AddHandler("sql", DeleteAuthInfo)
+}
+
+func GetUserByAuthInfo(query *m.GetUserByAuthInfoQuery) error {
+	user := &m.User{}
+	has := false
+	var err error
+
+	// Try to find the user by auth module and id first
+	if query.AuthModule != "" && query.AuthId != "" {
+		authQuery := &m.GetAuthInfoQuery{
+			AuthModule: query.AuthModule,
+			AuthId:     query.AuthId,
+		}
+
+		err = GetAuthInfo(authQuery)
+		// if user id was specified and doesn't match the user_auth entry, remove it
+		if err == nil && query.UserId != 0 && query.UserId != authQuery.UserAuth.UserId {
+			DeleteAuthInfo(&m.DeleteAuthInfoCommand{
```

I guess we ignore the error on purpose?

---

Yeah, I added some error logging there.